src/codegen/gencommon/closuresToClass.ml

```diff
@@ -413,6 +414,12 @@ let configure gen ft =
 		let cls = mk_class (get gen.gcurrent_class).cl_module path tfunc.tf_expr.epos in
 		if in_unsafe then cls.cl_meta <- (Meta.Unsafe,[],null_pos) :: cls.cl_meta;
 
+		(* Forward native type constraints meta *)
+		if Meta.has(Meta.Custom(":nativeTypeConstraints")) (get gen.gcurrent_class).cl_meta then begin
```

Comment:
I think it's better to also do the `is_hxgen` check here, or even better move `not is_hxgen && Meta.has` into a function and use it in both places. Also here you might want to check the platform, because this filter is also run on java (tho adding metadata should have no effect, but still).

Reply:
Ok thanks, I'll look into this (not sure where to put such a function though).